pkg/controllers/management/node/controller.go

```diff
@@ -763,10 +763,15 @@ func (m *Lifecycle) deleteV1Node(node *v3.Node) (runtime.Object, error) {
 		return node, err
 	}
 
+	ctx, cancel := context.WithTimeout(context.TODO(), 45*time.Second)
+	defer cancel()
 	err = userClient.K8sClient.CoreV1().Nodes().Delete(
-		context.TODO(), node.Status.NodeName, metav1.DeleteOptions{})
-	if !kerror.IsNotFound(err) {
-		return node, err
+		ctx, node.Status.NodeName, metav1.DeleteOptions{})
+	select {
+	case <-ctx.Done():
+		if !kerror.IsNotFound(err) && ctx.Err() != context.DeadlineExceeded {
+			return node, err
+		}
 	}
 
```

Comment:
This code blocks for 45 seconds until the context is cancelled due to the deadline being exceeded. Also since the context isn't being cancelled manually via `cancel()` until after the function returns, `ctx.Err()` will only ever be `context.DeadlineExceeded` which implies `return node, err` is unreachable, so this is functionally equivalent to:
```
userClient.K8sClient.CoreV1().Nodes().Delete(ctx, node.Status.NodeName, metav1.DeleteOptions{})
return node, nil
```

Adding this here so we have it, already discussed w/ @luthermonson